app/controllers/items_controller.rb

```diff
@@ -26,7 +26,14 @@ def index
     # end
   end
 
-  def show; end
+  def show
+    # link the show page to a new or existing request
+    if Request.find_by(item_id: @item.id, receiver_id: current_user.id, giver_id: @item.user.id)
+      @request = Request.find_by(item_id: @item.id, receiver_id: current_user.id, giver_id: @item.user.id)
+    else
+      @request = Request.new
+    end
+  end
```

Comment:
if there's an existing request, go to that @request. Otherwise, create a new @request